modules/alloydb-instance/variables.tf

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "automated_backup_policy" {
+  description = "The automated backup policy for this cluster. If no policy is provided then the default policy will be used. The default policy takes one backup a day, has a backup window of 1 hour, and retains backups for 14 days"
+  type = object({
+    location      = optional(string)
+    backup_window = optional(string)
+    enabled       = optional(bool)
+
+    weekly_schedule = optional(object({
+      days_of_week = optional(list(string))
+      start_times  = list(string)
+    })),
+
+    quantity_based_retention_count = optional(number)
+    time_based_retention_count     = optional(string)
+    labels                         = optional(map(string))
+    backup_encryption_key_name     = optional(string)
+  })
+  default = null
+}
+variable "cluster_id" {
+  description = "The ID of the alloydb cluster"
+  type        = string
+  validation {
+    condition     = can(regex("^[a-z0-9-]+$", var.cluster_id))
+    error_message = "ERROR: Cluster ID must contain only Letters(lowercase), number, and hyphen"
+  }
+}
+variable "cluster_display_name" {
```

Comment:
I would remove the `cluster_` prefix from variable names, except for id

Reply:
Need to differentiate b/w cluster and primary instance in AlloyDB. So I kept `cluster_` prefix specifically for the cluster.